src/renderers/testing/ReactShallowRenderer.js

```diff
@@ -100,26 +59,146 @@ class ReactShallowRenderer {
       element.type,
     );
 
-    // TODO Remove this wrapper if/when context parameter is removed
-    if (context && Object.keys(context).length) {
-      element = wrapElementWithContextProvider(element, context);
-    }
+    this._element = element;
 
-    this._renderer = ReactTestRenderer.create(element, {
-      createNodeMock: createShallowNodeMock(),
-    });
+    if (this._instance) {
+      this._rendered = updateClassComponent(
+        this._instance,
+        element.props,
+        context,
+      );
+    } else {
+      const prototype = element.type.prototype;
+
+      if (typeof prototype.render === 'function') {
```

Comment:
Sorry, I'm on the road and probably not making sense. What I'm trying to say is that the other examples check for something on the instance. But instance doesn't exist when you're "choosing" how to create it. There are only two places where we "choose" that (based on `type` alone), and these are the places we need to be consistent with since we also have just the `type`.

The overall algorithm is:

1. Check for prototype.isReactComponent.
2. If it exists, call type with new.
3. If it doesn't, call type without new.
4. Look at what type returned. If it's an object with render method, treat it as class instance from now on. Otherwise treat it as a functional component that returned an element.